src/main/java/blackjack/gametable/gambler/Dealer.java

```diff
@@ -0,0 +1,94 @@
+package blackjack.gametable.gambler;
+
+import blackjack.constant.MatchResult;
+import blackjack.gametable.card.Card;
+import blackjack.gametable.card.Cards;
+import java.util.List;
+
+public class Dealer extends Gambler {
+
+    public static final int MAX_SCORE = 21;
+    public static final int DEALER_HIT_THRESHOLD = 16;
+
+    @Override
+    public List<Card> openInitialCards() {
+        return cards.openDealerInitialCards();
+    }
+
+    public void updateBetAmounts(Players players) {
+        double dealerAmount = 0;
+
+        for (Player player : players.getPlayers()) {
+            MatchResult matchResult = compareTo(player);
+            dealerAmount += handleMatchResult(matchResult, player);
+        }
+
+        super.updateBetAmount(dealerAmount);
+    }
+
+    public boolean shouldDrawCard() {
+        return sumCardScores() <= DEALER_HIT_THRESHOLD;
+    }
+
+    private MatchResult compareTo(Player player) {
+        int dealerScore = sumCardScores();
+        int playerScore = player.sumCardScores();
+        Cards playerCards = player.getCards();
+
+        if (cards.isBlackjack() || playerCards.isBlackjack()) {
+            return calculateMatchResultWhenBlackjack(playerCards);
+        }
+
+        if (dealerScore > MAX_SCORE || playerScore > MAX_SCORE) {
+            return calculateMatchResultWhenOverBustStandard(playerScore);
+        }
+        return calculateMatchResult(dealerScore, playerScore);
+    }
+
+    private double handleMatchResult(MatchResult matchResult, Player player) {
+        double betAmount = player.getBetAmount();
+
+        if (matchResult == MatchResult.BLACKJACK_WIN) {
+            double blackjackAmount = betAmount * 1.5;
+            player.updateBetAmount(blackjackAmount);
+            return -blackjackAmount;
+        }
+        if (matchResult == MatchResult.WIN) {
+            return -betAmount;
+        }
+        if (matchResult == MatchResult.LOSE) {
+            player.updateBetAmount(-betAmount);
+            return betAmount;
+        }
+        player.updateBetAmount(0);
+        return 0;
```

Comment:
각각의 MatchResult 요소마다 `배팅 배당`(?)이 다른 것으로 보이는데요. 
배팅 배당율을 MatchResult의 상태 값으로 두면 어떨까요? 

그렇다면 분기문을 없앨 수도 있지 않을까요? 

```suggestion
       double profit = betAmount * 1.5;
       player.updateBetAmount(profit);
       return profit;
```